### PR TITLE
[stable/memcached] Add security context to memcached chart

### PR DESCRIPTION
**Pull request by JorisAndrade**
## What this PR does / why we need it:

This PR adds the securityContext option to be able to run this container as non-root. 
Right now if you have a PSP in your namespace with 

```
  runAsUser:
    rule: MustRunAsNonRoot
```

memcached will fail to start with `Error: container has runAsNonRoot and image has non-numeric user (memcache), cannot verify user is non-root`

#### Special notes for your reviewer:
This is my first PR to this repo so feel free to ping me if something is missing!
I'm not sure about the `version` for example, should I bump patch or minor?

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by JorisAndrade** [comment]
@olemarkus any chance you can have a look to this?

File: stable/memcached/README.md [hunk 1]
```diff
@@ -40,25 +40,28 @@ The command removes all the Kubernetes components associated with the chart and
...
+| `extraContainers`          | Container sidecar definition(s) as string | Un-set                                        |
+| `extraVolumes`             | Volume definitions to add as string | Un-set                                              |
+| `kind`                     | Install as StatefulSet or Deployment | StatefulSet                                        |
+| `podAnnotations`           | Map of annotations to add to the pod(s) | `{}`                                            |
+| `podLabels`                | Custom Labels to be applied to statefulset | Un-set                                       |
+| `nodeSelector`             | Simple pod scheduling control | `{}`                                                      |
+| `tolerations`              | Allow or deny specific node taints | `{}`                                                 |
+| `affinity`                 | Advanced pod scheduling control | `{}`                                                    |
+| `securityContext.enabled`  | Enable security context (both redis master and slave pods) | `true`                       |
```
Comment (olemarkus) [inline stable/memcached/README.md]: I guess this is a copy/paste issue

File: stable/memcached/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: memcached
-version: 2.8.3
+version: 2.8.4
```
Comment (olemarkus) [inline stable/memcached/Chart.yaml]: Need to bump minor version (2.9.0)

File: stable/memcached/values.yaml [hunk 3]
```diff
@@ -61,6 +61,13 @@ tolerations: {}
 ## ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
 affinity: {}
 
+## Memcached pod Security Context
+## ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
+securityContext:
+  enabled: true
+  fsGroup: 11211
```
Comment (olemarkus) [inline stable/memcached/values.yaml]: Why these uid/guids? E.g 1001 is a more common value here.

**Review by olemarkus — CHANGES_REQUESTED** [review]

File: stable/memcached/values.yaml (hunk 3 above)
Comment (JorisAndrade) [reply stable/memcached/values.yaml]: Only because it was used here -> https://github.com/docker-library/memcached/blob/master/alpine/Dockerfile#L4

Comment (JorisAndrade) [reply stable/memcached/values.yaml]: but I can put `1001` :)

**Comment by JorisAndrade** [comment]
@olemarkus addressed your comments but feel free to ping me if something else is needed :)

**Pull request merged**